File: clients/node/sdk/models/relationship.ts
import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class Relationship implements AdditionalDataHolder, Parsable {
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
    private _additionalData: Record<string, unknown>;
    /** The blocked_by property */
    private _blocked_by?: boolean | undefined;
    /** The blocking property */
    private _blocking?: boolean | undefined;
    /** The domain_blocking property */
    private _domain_blocking?: boolean | undefined;
    /** The endorsed property */
    private _endorsed?: boolean | undefined;
    /** The followed_by property */
    private _followed_by?: boolean | undefined;
    /** The following property */
    private _following?: boolean | undefined;
    /** The id property */
    private _id?: string | undefined;
    /** The languages property */
    private _languages?: string[] | undefined;
    /** The muting property */
    private _muting?: boolean | undefined;
    /** The muting_notifications property */
    private _muting_notifications?: boolean | undefined;
    /** The note property */
    private _note?: string | undefined;
    /** The notifying property */
    private _notifying?: boolean | undefined;
    /** The requested property */
    private _requested?: boolean | undefined;
    /** The showing_reblogs property */
    private _showing_reblogs?: boolean | undefined;
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    public get additionalData() {
        return this._additionalData;
    };
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public set additionalData(value: Record<string, unknown>) {
        this._additionalData = value;
    };
    /**
     * Gets the blocked_by property value. The blocked_by property
     * @returns a boolean
     */
    public get blocked_by() {
        return this._blocked_by;
    };
    /**
     * Sets the blocked_by property value. The blocked_by property
     * @param value Value to set for the blocked_by property.
     */
    public set blocked_by(value: boolean | undefined) {
        this._blocked_by = value;
    };
    /**
     * Gets the blocking property value. The blocking property
     * @returns a boolean
     */
    public get blocking() {
        return this._blocking;
    };
    /**
     * Sets the blocking property value. The blocking property
     * @param value Value to set for the blocking property.
     */
    public set blocking(value: boolean | undefined) {
        this._blocking = value;
    };
    /**
     * Instantiates a new Relationship and sets the default values.
     */
    public constructor() {
        this._additionalData = {};
    };
    /**
     * Gets the domain_blocking property value. The domain_blocking property
     * @returns a boolean
     */
    public get domain_blocking() {
        return this._domain_blocking;
    };
    /**
     * Sets the domain_blocking property value. The domain_blocking property
     * @param value Value to set for the domain_blocking property.
     */
    public set domain_blocking(value: boolean | undefined) {
        this._domain_blocking = value;
    };
    /**
     * Gets the endorsed property value. The endorsed property
     * @returns a boolean
     */
    public get endorsed() {
        return this._endorsed;
    };
    /**
     * Sets the endorsed property value. The endorsed property
     * @param value Value to set for the endorsed property.
     */
    public set endorsed(value: boolean | undefined) {
        this._endorsed = value;
    };
    /**
     * Gets the followed_by property value. The followed_by property
     * @returns a boolean
     */
    public get followed_by() {
        return this._followed_by;
    };
    /**
     * Sets the followed_by property value. The followed_by property
     * @param value Value to set for the followed_by property.
     */
    public set followed_by(value: boolean | undefined) {
        this._followed_by = value;
    };
    /**
     * Gets the following property value. The following property
     * @returns a boolean
     */
    public get following() {
        return this._following;
    };
    /**
     * Sets the following property value. The following property
     * @param value Value to set for the following property.
     */
    public set following(value: boolean | undefined) {
        this._following = value;
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {
            "blocked_by": n => { this.blocked_by = n.getBooleanValue(); },
            "blocking": n => { this.blocking = n.getBooleanValue(); },
            "domain_blocking": n => { this.domain_blocking = n.getBooleanValue(); },
            "endorsed": n => { this.endorsed = n.getBooleanValue(); },
            "followed_by": n => { this.followed_by = n.getBooleanValue(); },
            "following": n => { this.following = n.getBooleanValue(); },
            "id": n => { this.id = n.getStringValue(); },
            "languages": n => { this.languages = n.getCollectionOfPrimitiveValues<string>(); },
            "muting": n => { this.muting = n.getBooleanValue(); },
            "muting_notifications": n => { this.muting_notifications = n.getBooleanValue(); },
            "note": n => { this.note = n.getStringValue(); },
            "notifying": n => { this.notifying = n.getBooleanValue(); },
            "requested": n => { this.requested = n.getBooleanValue(); },
            "showing_reblogs": n => { this.showing_reblogs = n.getBooleanValue(); },
        };
    };
    /**
     * Gets the id property value. The id property
     * @returns a string
     */
    public get id() {
        return this._id;
    };
    /**
     * Sets the id property value. The id property
     * @param value Value to set for the id property.
     */
    public set id(value: string | undefined) {
        this._id = value;
    };
    /**
     * Gets the languages property value. The languages property
     * @returns a string
     */
    public get languages() {
        return this._languages;
    };
    /**
     * Sets the languages property value. The languages property
     * @param value Value to set for the languages property.
     */
    public set languages(value: string[] | undefined) {
        this._languages = value;
    };
    /**
     * Gets the muting property value. The muting property
     * @returns a boolean
     */
    public get muting() {
        return this._muting;
    };
    /**
     * Sets the muting property value. The muting property
     * @param value Value to set for the muting property.
     */
    public set muting(value: boolean | undefined) {
        this._muting = value;
    };
    /**
     * Gets the muting_notifications property value. The muting_notifications property
     * @returns a boolean
     */
    public get muting_notifications() {
        return this._muting_notifications;
    };
    /**
     * Sets the muting_notifications property value. The muting_notifications property
     * @param value Value to set for the muting_notifications property.
     */
    public set muting_notifications(value: boolean | undefined) {
        this._muting_notifications = value;
    };
    /**
     * Gets the note property value. The note property
     * @returns a string
     */
    public get note() {
        return this._note;
    };
    /**
     * Sets the note property value. The note property
     * @param value Value to set for the note property.
     */
    public set note(value: string | undefined) {
        this._note = value;
    };
    /**
     * Gets the notifying property value. The notifying property
     * @returns a boolean
     */
    public get notifying() {
        return this._notifying;
    };
    /**
     * Sets the notifying property value. The notifying property
     * @param value Value to set for the notifying property.
     */
    public set notifying(value: boolean | undefined) {
        this._notifying = value;
    };
    /**
     * Gets the requested property value. The requested property
     * @returns a boolean
     */
    public get requested() {
        return this._requested;
    };
    /**
     * Sets the requested property value. The requested property
     * @param value Value to set for the requested property.
     */
    public set requested(value: boolean | undefined) {
        this._requested = value;
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        writer.writeBooleanValue("blocked_by", this.blocked_by);
        writer.writeBooleanValue("blocking", this.blocking);
        writer.writeBooleanValue("domain_blocking", this.domain_blocking);
        writer.writeBooleanValue("endorsed", this.endorsed);
        writer.writeBooleanValue("followed_by", this.followed_by);
        writer.writeBooleanValue("following", this.following);
        writer.writeStringValue("id", this.id);
        writer.writeCollectionOfPrimitiveValues<string>("languages", this.languages);
        writer.writeBooleanValue("muting", this.muting);
        writer.writeBooleanValue("muting_notifications", this.muting_notifications);
        writer.writeStringValue("note", this.note);
        writer.writeBooleanValue("notifying", this.notifying);
        writer.writeBooleanValue("requested", this.requested);
        writer.writeBooleanValue("showing_reblogs", this.showing_reblogs);
        writer.writeAdditionalData(this.additionalData);
    };
    /**
     * Gets the showing_reblogs property value. The showing_reblogs property
     * @returns a boolean
     */
    public get showing_reblogs() {
        return this._showing_reblogs;
    };
    /**
     * Sets the showing_reblogs property value. The showing_reblogs property
     * @param value Value to set for the showing_reblogs property.
     */
    public set showing_reblogs(value: boolean | undefined) {
        this._showing_reblogs = value;
    };
}

File: clients/python/sdk/models/relationship.py
from __future__ import annotations
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, Union

class Relationship(AdditionalDataHolder, Parsable):
    @property
    def additional_data(self,) -> Dict[str, Any]:
        """
        Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Returns: Dict[str, Any]
        """
        return self._additional_data
    
    @additional_data.setter
    def additional_data(self,value: Dict[str, Any]) -> None:
        """
        Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Args:
            value: Value to set for the AdditionalData property.
        """
        self._additional_data = value
    
    @property
    def blocked_by(self,) -> Optional[bool]:
        """
        Gets the blocked_by property value. The blocked_by property
        Returns: Optional[bool]
        """
        return self._blocked_by
    
    @blocked_by.setter
    def blocked_by(self,value: Optional[bool] = None) -> None:
        """
        Sets the blocked_by property value. The blocked_by property
        Args:
            value: Value to set for the blocked_by property.
        """
        self._blocked_by = value
    
    @property
    def blocking(self,) -> Optional[bool]:
        """
        Gets the blocking property value. The blocking property
        Returns: Optional[bool]
        """
        return self._blocking
    
    @blocking.setter
    def blocking(self,value: Optional[bool] = None) -> None:
        """
        Sets the blocking property value. The blocking property
        Args:
            value: Value to set for the blocking property.
        """
        self._blocking = value
    
    def __init__(self,) -> None:
        """
        Instantiates a new Relationship and sets the default values.
        """
        # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        self._additional_data: Dict[str, Any] = {}

        # The blocked_by property
        self._blocked_by: Optional[bool] = None
        # The blocking property
        self._blocking: Optional[bool] = None
        # The domain_blocking property
        self._domain_blocking: Optional[bool] = None
        # The endorsed property
        self._endorsed: Optional[bool] = None
        # The followed_by property
        self._followed_by: Optional[bool] = None
        # The following property
        self._following: Optional[bool] = None
        # The id property
        self._id: Optional[str] = None
        # The languages property
        self._languages: Optional[List[str]] = None
        # The muting property
        self._muting: Optional[bool] = None
        # The muting_notifications property
        self._muting_notifications: Optional[bool] = None
        # The note property
        self._note: Optional[str] = None
        # The notifying property
        self._notifying: Optional[bool] = None
        # The requested property
        self._requested: Optional[bool] = None
        # The showing_reblogs property
        self._showing_reblogs: Optional[bool] = None
    
    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> Relationship:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: Relationship
        """
        if parse_node is None:
            raise Exception("parse_node cannot be undefined")
        return Relationship()
    
    @property
    def domain_blocking(self,) -> Optional[bool]:
        """
        Gets the domain_blocking property value. The domain_blocking property
        Returns: Optional[bool]
        """
        return self._domain_blocking
    
    @domain_blocking.setter
    def domain_blocking(self,value: Optional[bool] = None) -> None:
        """
        Sets the domain_blocking property value. The domain_blocking property
        Args:
            value: Value to set for the domain_blocking property.
        """
        self._domain_blocking = value
    
    @property
    def endorsed(self,) -> Optional[bool]:
        """
        Gets the endorsed property value. The endorsed property
        Returns: Optional[bool]
        """
        return self._endorsed
    
    @endorsed.setter
    def endorsed(self,value: Optional[bool] = None) -> None:
        """
        Sets the endorsed property value. The endorsed property
        Args:
            value: Value to set for the endorsed property.
        """
        self._endorsed = value
    
    @property
    def followed_by(self,) -> Optional[bool]:
        """
        Gets the followed_by property value. The followed_by property
        Returns: Optional[bool]
        """
        return self._followed_by
    
    @followed_by.setter
    def followed_by(self,value: Optional[bool] = None) -> None:
        """
        Sets the followed_by property value. The followed_by property
        Args:
            value: Value to set for the followed_by property.
        """
        self._followed_by = value
    
    @property
    def following(self,) -> Optional[bool]:
        """
        Gets the following property value. The following property
        Returns: Optional[bool]
        """
        return self._following
    
    @following.setter
    def following(self,value: Optional[bool] = None) -> None:
        """
        Sets the following property value. The following property
        Args:
            value: Value to set for the following property.
        """
        self._following = value
    
    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
            "blocked_by": lambda n : setattr(self, 'blocked_by', n.get_bool_value()),
            "blocking": lambda n : setattr(self, 'blocking', n.get_bool_value()),
            "domain_blocking": lambda n : setattr(self, 'domain_blocking', n.get_bool_value()),
            "endorsed": lambda n : setattr(self, 'endorsed', n.get_bool_value()),
            "followed_by": lambda n : setattr(self, 'followed_by', n.get_bool_value()),
            "following": lambda n : setattr(self, 'following', n.get_bool_value()),
            "id": lambda n : setattr(self, 'id', n.get_str_value()),
            "languages": lambda n : setattr(self, 'languages', n.get_collection_of_primitive_values(str)),
            "muting": lambda n : setattr(self, 'muting', n.get_bool_value()),
            "muting_notifications": lambda n : setattr(self, 'muting_notifications', n.get_bool_value()),
            "note": lambda n : setattr(self, 'note', n.get_str_value()),
            "notifying": lambda n : setattr(self, 'notifying', n.get_bool_value()),
            "requested": lambda n : setattr(self, 'requested', n.get_bool_value()),
            "showing_reblogs": lambda n : setattr(self, 'showing_reblogs', n.get_bool_value()),
        }
        return fields
    
    @property
    def id(self,) -> Optional[str]:
        """
        Gets the id property value. The id property
        Returns: Optional[str]
        """
        return self._id
    
    @id.setter
    def id(self,value: Optional[str] = None) -> None:
        """
        Sets the id property value. The id property
        Args:
            value: Value to set for the id property.
        """
        self._id = value
    
    @property
    def languages(self,) -> Optional[List[str]]:
        """
        Gets the languages property value. The languages property
        Returns: Optional[List[str]]
        """
        return self._languages
    
    @languages.setter
    def languages(self,value: Optional[List[str]] = None) -> None:
        """
        Sets the languages property value. The languages property
        Args:
            value: Value to set for the languages property.
        """
        self._languages = value
    
    @property
    def muting(self,) -> Optional[bool]:
        """
        Gets the muting property value. The muting property
        Returns: Optional[bool]
        """
        return self._muting
    
    @muting.setter
    def muting(self,value: Optional[bool] = None) -> None:
        """
        Sets the muting property value. The muting property
        Args:
            value: Value to set for the muting property.
        """
        self._muting = value
    
    @property
    def muting_notifications(self,) -> Optional[bool]:
        """
        Gets the muting_notifications property value. The muting_notifications property
        Returns: Optional[bool]
        """
        return self._muting_notifications
    
    @muting_notifications.setter
    def muting_notifications(self,value: Optional[bool] = None) -> None:
        """
        Sets the muting_notifications property value. The muting_notifications property
        Args:
            value: Value to set for the muting_notifications property.
        """
        self._muting_notifications = value
    
    @property
    def note(self,) -> Optional[str]:
        """
        Gets the note property value. The note property
        Returns: Optional[str]
        """
        return self._note
    
    @note.setter
    def note(self,value: Optional[str] = None) -> None:
        """
        Sets the note property value. The note property
        Args:
            value: Value to set for the note property.
        """
        self._note = value
    
    @property
    def notifying(self,) -> Optional[bool]:
        """
        Gets the notifying property value. The notifying property
        Returns: Optional[bool]
        """
        return self._notifying
    
    @notifying.setter
    def notifying(self,value: Optional[bool] = None) -> None:
        """
        Sets the notifying property value. The notifying property
        Args:
            value: Value to set for the notifying property.
        """
        self._notifying = value
    
    @property
    def requested(self,) -> Optional[bool]:
        """
        Gets the requested property value. The requested property
        Returns: Optional[bool]
        """
        return self._requested
    
    @requested.setter
    def requested(self,value: Optional[bool] = None) -> None:
        """
        Sets the requested property value. The requested property
        Args:
            value: Value to set for the requested property.
        """
        self._requested = value
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if writer is None:
            raise Exception("writer cannot be undefined")
        writer.write_bool_value("blocked_by", self.blocked_by)
        writer.write_bool_value("blocking", self.blocking)
        writer.write_bool_value("domain_blocking", self.domain_blocking)
        writer.write_bool_value("endorsed", self.endorsed)
        writer.write_bool_value("followed_by", self.followed_by)
        writer.write_bool_value("following", self.following)
        writer.write_str_value("id", self.id)
        writer.write_collection_of_primitive_values("languages", self.languages)
        writer.write_bool_value("muting", self.muting)
        writer.write_bool_value("muting_notifications", self.muting_notifications)
        writer.write_str_value("note", self.note)
        writer.write_bool_value("notifying", self.notifying)
        writer.write_bool_value("requested", self.requested)
        writer.write_bool_value("showing_reblogs", self.showing_reblogs)
        writer.write_additional_data_value(self.additional_data)
    
    @property
    def showing_reblogs(self,) -> Optional[bool]:
        """
        Gets the showing_reblogs property value. The showing_reblogs property
        Returns: Optional[bool]
        """
        return self._showing_reblogs
    
    @showing_reblogs.setter
    def showing_reblogs(self,value: Optional[bool] = None) -> None:
        """
        Sets the showing_reblogs property value. The showing_reblogs property
        Args:
            value: Value to set for the showing_reblogs property.
        """
        self._showing_reblogs = value
    

